src/pages/exam_attendance_report.js

```diff
@@ -1,51 +1,80 @@
-import { Box, Typography, Button, Grid } from '@mui/material';
+import { Box, Typography, Button, Grid, CircularProgress } from '@mui/material';
 import IconButton from '@mui/material/IconButton';
 import ArrowBackIosIcon from '@mui/icons-material/ArrowBackIos';
 import ExamAttendanceTable from '../components/exam_attendance_table';
 import AttendanceSummary from '../components/attendance_summary';
 import { useNavigate } from 'react-router-dom';
-import NavBar from '../components/Navbar';
+import { useAppContext } from "../context/appContext";
+import { useEffect, useState } from 'react';
+import { useParams } from 'react-router-dom';
 
-function createData(name, index, attendance, paperCollected) {
-  return { name, index, attendance, paperCollected };
-}
-
-const data = [
-  createData('Kasun Kasun', "190300A", "08:05 AM", "10.05 AM"),
-  createData('Nimal Sirisena', "190890P", "08:05 AM", ""),
-  createData('Pramila Perera', "190000X", "", "08:05 AM"),
-  createData('Kamal Induwara', "190215X", "", "08:05 AM"),
-  createData('Kasun Kasun', "190300A", "08:05 AM", ""),
-  createData('Nimal Sirisena', "190000X", "08:05 AM", ""),
-  createData('Pramila Perera', "190890P", "", "08:05 AM"),
-  createData('Kamal Induwara', "190300A", "", "08:05 AM"),
-  createData('Kasun Kasun', "190000X", "08:05 AM", ""),
-  createData('Nimal Sirisena', "190215X", "08:05 AM", ""),
-  createData('Pramila Perera', "190000X", "", "08:05 AM"),
-  createData('Kamal Induwara', "190890P", "", "08:05 AM"),
-  createData('Kasun Kasun', "190000X", "08:05 AM", ""),
-  createData('Nimal Sirisena', "190890P", "08:05 AM", ""),
-  createData('Pramila Perera', "190215X", "", "08:05 AM"),
-  createData('Kamal Induwara', "190890P", "", "08:05 AM"),
-];
-
-function getTotalCount() {
+function getTotalCount(data) {
     return data.length;
 }
 
-function getPresentCount() { 
+function getPresentCount(data) { 
     let result = 0;
     for (let i = 0; i < data.length; i++) {
         const row = data[i];
-        if (row.attendance !== "" && row.attendance !== null) {
+        if (row.present) {
             result = result + 1;
         }
     }
     return result;
 }
 
 const Report = () => {
-  const navigate = useNavigate()
+  const navigate = useNavigate();
+  const { examid } = useParams();
+  const { authFetch } = useAppContext();
+  const [attendance, setAttendance] = useState([]);
+  const [exam, setExam] = useState({});
+  const [schedule, setSchedule] = useState();
+  const [isLoadingExam, setIsLoadingExam] = useState(true);
+  const [isLoadingAttendance, setIsLoadingAttendance] = useState(true);
```

Comment:
Can we change this initial state to false and set it to true at the begining of the function that fetching data and set to false after function complete.

Reply:
Done